### PR TITLE
[ESP32] Added config option to disable and de-init BLE post commissioning

### PR DESCRIPTION
**Pull request by jadhavrohit924**
#### Problem
* Added config option to disable and de-init BLE after commissioning is successful.

#### Change overview
Total disable and deinit BLE and reclaim all its memory

#### Testing
Manually tested

### CODE REVIEW

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 1]
```diff
@@ -114,9 +117,24 @@ void DeviceCallbacks::DeviceEventCallback(const ChipDeviceEvent * event, intptr_
...
-        break;
+#if CONFIG_BT_NIMBLE_ENABLED && CONFIG_DEINIT_BLE_ON_COMMISSIONING_COMPLETE
+        esp_err_t err = ESP_OK;
+        int ret       = nimble_port_stop();
+        if (ret == 0)
+        {
+            nimble_port_deinit();
+            err = esp_nimble_hci_and_controller_deinit();
+            err = esp_bt_mem_release(ESP_BT_MODE_BLE);
+        }
```
Comment (andy31415) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:130]: what if ret is not 0? Should we log something or is that normal?

**Review by andy31415 — APPROVED** [review]

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 2]
```diff
@@ -114,9 +117,27 @@ void DeviceCallbacks::DeviceEventCallback(const ChipDeviceEvent * event, intptr_
...
+    case DeviceEventType::kCommissioningComplete: {
         ESP_LOGI(TAG, "Commissioning complete");
-        break;
+#if CONFIG_BT_NIMBLE_ENABLED && CONFIG_DEINIT_BLE_ON_COMMISSIONING_COMPLETE
+        int ret = nimble_port_stop();
+        if (ret == 0)
+        {
+            nimble_port_deinit();
+            esp_err_t err = esp_nimble_hci_and_controller_deinit();
+            err           = esp_bt_mem_release(ESP_BT_MODE_BLE);
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:128]: `err += esp_bt_mem_release(ESP_BT_MODE_BLE);` ?

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 3]
```diff
@@ -114,9 +117,27 @@ void DeviceCallbacks::DeviceEventCallback(const ChipDeviceEvent * event, intptr_
...
+            if (err == ESP_OK)
+            {
+                ESP_LOGI(TAG, "BLE deinit successful");
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:131]: BLE deinit successful and memory reclaimed?

File: examples/lighting-app/esp32/main/DeviceCallbacks.cpp [hunk 4]
```diff
@@ -66,9 +69,27 @@ void DeviceCallbacks::DeviceEventCallback(const ChipDeviceEvent * event, intptr_
...
+    case DeviceEventType::kCommissioningComplete: {
         ESP_LOGI(TAG, "Commissioning complete");
-        break;
+#if CONFIG_BT_NIMBLE_ENABLED && CONFIG_DEINIT_BLE_ON_COMMISSIONING_COMPLETE
+        int ret = nimble_port_stop();
+        if (ret == 0)
+        {
+            nimble_port_deinit();
+            esp_err_t err = esp_nimble_hci_and_controller_deinit();
+            err           = esp_bt_mem_release(ESP_BT_MODE_BLE);
```
Comment (dhrishi) [inline examples/lighting-app/esp32/main/DeviceCallbacks.cpp:80]: Similar changes as mentioned above

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp (hunk 2 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp]: done

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp (hunk 3 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp]: done

**Review by dhrishi — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**